File: Civitas_Ruby/lib/sorpresa_salirCarcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'sorpresa.rb'
require_relative 'jugador.rb'
require_relative 'diario.rb'
module Civitas
  class Sorpresa_salirCarcel < Sorpresa
    def initialize(mazo,texto)
      super(texto)
      @mazo = mazo
    end
    
    def aplicarAJugador(actual, todos)
      if(jugadorCorrecto(actual, todos))
        informe(actual, todos)
        
        salvoconducto = false;
        for i in todos
          if(!salvoconducto)
            salvoconducto = i.tieneSalvoconducto()
          end
        end
      
        if(!salvoconducto)
          todos[actual].obtenerSalvoconducto(self)
          self.salirDelMazo();
        end
      end
    end
    
    def salirDelMazo()
      @mazo.inhabilitarCartaEspecial(self)
    end
    
    
    def usada()
      @mazo.habilitarCartaEspecial(self)
    end
    
    def informe(actual, todos)
      Diario.instance().ocurre_evento(todos[actual].to_s + " se ve afectado por una carta de tipo salirCarcel y " + to_s())
    end
    
    
  end
end

File: Civitas_GUI/src/civitas/Sorpresa_salirCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;
import java.util.ArrayList;
/**
 *
 * @author francisco
 */
class Sorpresa_salirCarcel extends Sorpresa{
    private MazoSorpresas mazo;
    
    Sorpresa_salirCarcel(MazoSorpresas mazo1){
        super();
        this.mazo = mazo1;
    }
    
    
    void salirDelMazo(){
            mazo.inhabilitarCartaEspecial(this);
    }
    
    void usada(){
            mazo.habilitarCartaEspecial(this);
    }
    
    @Override
    protected void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if (super.jugadorCorrecto(actual,todos)){
            this.informe(actual, todos);
            
            Boolean salvoConducto = false;
            for (int i = 0; i < todos.size() && !salvoConducto; i++)
                salvoConducto = todos.get(i).tieneSalvoconducto();
            
            if (!salvoConducto){
                todos.get(actual).obtenerSalvoconducto(this);  
                this.salirDelMazo();
            }
            
        }
    }
    
    @Override
    protected void informe(int actual, ArrayList<Jugador> todos){
        if(super.jugadorCorrecto(actual, todos))
            Diario.getInstance().ocurreEvento(todos.get(actual).getNombre() + " se ve afectado por una sorpresa de tipo salirCarcel ");
    }
}
